tests/std/tests/P2093R14_formatted_output/test.cpp

```diff
@@ -0,0 +1,642 @@
+    } else {
+        test::WinSemaphore hello_semaphore{};
+        test::WinSemaphore goodbye_semaphore{};
+
+        // This will receive the child process ID.
+        PROCESS_INFORMATION process_information{};
+
+        {
+            // Get the absolute path of our executable.
+            // This assumes that our test infrastructure doesn't use long paths.
+            wstring module_filename;
+            module_filename.resize_and_overwrite(MAX_PATH, [](wchar_t* const ptr, const size_t n) {
+                // resize_and_overwrite() prepares a buffer of size n + 1.
+                // GetModuleFileNameW() takes that buffer size.
+                const DWORD len = GetModuleFileNameW(nullptr, ptr, static_cast<DWORD>(n + 1));
+                assert(len != 0);
+                return len;
+            });
+
+            // CreateProcessW() requires the command line to be non-const, so we pass command_line.data() below.
+            // We use quotes to defend against module_filename containing spaces.
+            wstring command_line = format(LR"("{}" --child {} {})", module_filename, hello_semaphore.to_uintptr(),
+                goodbye_semaphore.to_uintptr());
+
+            // The entire purpose of this code is to hide the child process's console window,
+            // to avoid rapid flickering during test runs.
+            STARTUPINFOW startup_info{
+                .cb          = sizeof(STARTUPINFOW),
+                .dwFlags     = STARTF_USESHOWWINDOW,
+                .wShowWindow = SW_HIDE,
+            };
+
+            constexpr BOOL inherit_handles = TRUE;
+
+            // https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags
+            // "The new process has a new console, instead of inheriting its parent's console (the default)."
+            constexpr DWORD creation_flags = CREATE_NEW_CONSOLE;
+
+            const BOOL create_process_succeeded = CreateProcessW(module_filename.c_str(), command_line.data(), nullptr,
+                nullptr, inherit_handles, creation_flags, nullptr, nullptr, &startup_info, &process_information);
+            assert(create_process_succeeded);
+        }
+
+        // Wait for the child process to be ready before attaching to its console.
+        hello_semaphore.acquire();
+
+        {
+            // We want to do things like alter the output code page of our console, but we don't want
+            // to affect other tests which run concurrently. So, we want to detach this process from
+            // its current console, attach to the child process's console, and modify that one instead.
+
+            const BOOL free_console_succeeded = FreeConsole();
+            assert(free_console_succeeded);
+
+            const BOOL attach_console_succeeded = AttachConsole(process_information.dwProcessId);
+            assert(attach_console_succeeded);
+
+            const BOOL set_console_output_cp_succeeded = SetConsoleOutputCP(CP_UTF8);

```

Comment:
the `<print>` code needs to work even if the console's output codepage is not UTF-8; I think we should test this with at least one non-utf-8 codepage.